editor/state.js

```diff
@@ -208,17 +227,56 @@ export function isSidebarOpened( state = false, action ) {
 }
 
 /**
+ * Reducer returning current network request state (whether a request to the WP
+ * REST API is in progress, successful, or failed).
+ *
+ * @param  {string} state  Current state
+ * @param  {Object} action Dispatched action
+ * @return {string}        Updated state
+ */
+export function saving( state = {}, action ) {
+	switch ( action.type ) {
+		case 'POST_UPDATE_REQUEST':
+			return {
```

Comment:
I tried this out, and it ended up being longer and (IMO) harder to read:

```js
export const saving = combineReducers( {
	requesting: ( state = false, action ) => {
		switch ( action.type ) {
			case 'REQUEST_POST_UPDATE':
				return true;
			case 'REQUEST_POST_UPDATE_SUCCESS':
			case 'REQUEST_POST_UPDATE_FAILURE':
				return false;
		}
		return state;
	},
	successful: ( state = false, action ) => {
		switch ( action.type ) {
			case 'REQUEST_POST_UPDATE_SUCCESS':
				return true;
			case 'REQUEST_POST_UPDATE_FAILURE':
				return false;
		}
		return state;
	},
	error: ( state = false, action ) => {
		switch ( action.type ) {
			case 'REQUEST_POST_UPDATE_SUCCESS':
				return false;
			case 'REQUEST_POST_UPDATE_FAILURE':
				return true;
		}
		return state;
	},
	isNew: ( state = false, action ) => {
		switch ( action.type ) {
			case 'REQUEST_POST_UPDATE':
			case 'REQUEST_POST_UPDATE_SUCCESS':
			case 'REQUEST_POST_UPDATE_FAILURE':
				return action.isNew;
		}
		return state;
	},
} );
```

In this case, at least, I prefer the explicitness of "here are 3 action types and here is how each property changes in response".  I think updating state references and possible re-renders when a network request is fired or completed is OK.